### PR TITLE
Add support for extra fields

### PR DESCRIPTION
**Pull request by d-miketa**
Jira boards are very customisable. This PR adds the ability to declare extra UDA fields from `bugwarriorrc`.

https://github.com/ralphbean/bugwarrior/issues/860

### CODE REVIEW

File: bugwarrior/services/jira.py
Comment (ryneeverett) [inline bugwarrior/services/jira.py]: To me this looks weird -- can we please consistently include or omit an underscore in these variables?

Personally I'd just add `**self.get_extrafields(),` to the end of the existing dict, eliminating the need for variables. But I leave that to your discretion.

File: bugwarrior/services/jira.py
Comment (ryneeverett) [inline bugwarrior/services/jira.py]: Shouldn't this be `'.'.join(keys)` since we don't know which key caused the `KeyError`? (I'm assuming that's the only usage of this method.)

File: bugwarrior/services/jira.py
Comment (ryneeverett) [inline bugwarrior/services/jira.py:60]: Couldn't we use a standard library `dataclasses.dataclass` here? I guess we still support python3.6, which doesn't include dataclasses, so maybe just add a comment that we can switch once we drop 3.6?

**Review by ryneeverett — APPROVED** [review]
I'd like to see my comments addressed before merging but they're fairly trivial in the scope of this patch.

File: bugwarrior/services/jira.py
Comment (d-miketa) [reply bugwarrior/services/jira.py]: Yeah I agree and have fixed it.

The reason for this format was because I originally wanted to implement this for fields of the form `customfield_12345`, so I called the variable `customfields`. Then I realised I can target _any_ field, so switched to the more generic `extrafields`. But of course now that no longer makes sense!

**Comment by ryneeverett** [comment]
Merged in 55db4c4a212896f5807ea61ab95ace434296f080.

File: bugwarrior/services/jira.py
Comment (ryneeverett) [inline bugwarrior/services/jira.py:98]: Note that in the rebase I changed this to `extra_fields: typing.Optional[JiraExtraFields] = None` for consistency with other non-required fields with empty defaults.

**Pull request closed without merging**
